include/nanobind/stl/complex.h

```diff
@@ -0,0 +1,53 @@
+/*
+    nanobind/stl/complex.h: type caster for std::complex<...>
+
+    Copyright (c) 2023 Degottex Gilles
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/nanobind.h>
+#include <complex>
+
+#include <iostream>
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+NAMESPACE_BEGIN(detail)
+
+template <typename T1> struct type_caster<std::complex<T1>> {
+
+    // Sub type caster
+    // TODO This definition seems useless, but I can't find a neat way to replace `Caster::Name` below.
+    using Caster = make_caster<T1>; 
+    NB_TYPE_CASTER(std::complex<T1>, const_name("complex [") + Caster::Name + const_name("]") )
```

Comment:
Can remove ``Caster::Name`` here and just denote the type name as `"complex"` (anything else would be a pretty weird type signature in Python).

Reply:
Those type annotations are intended to make sense in Python, and there isn't a separate single precision float object.

Reply:
done